### PR TITLE
Tweak a few samples

### PR DESCRIPTION
**Pull request by stephentoub**
A little cleanup in ListBot sample, plus fix thread-unsafe use of Random in a few other samples

### CODE REVIEW

File: dotnet/samples/04.e.twentyQuestions/ResponseBuilder.cs
Comment (corinagum) [inline dotnet/samples/04.e.twentyQuestions/ResponseBuilder.cs:143]: I'm not a `dotnet` dev; could you explain this one to me? :)

**Review by corinagum — APPROVED** [review]
One question, otherwise LGTM! We want at least one `dotnet` codeowner to approve before merging. I've assigned this internally. Thanks!

File: dotnet/samples/04.e.twentyQuestions/ResponseBuilder.cs
Comment (stephentoub) [reply dotnet/samples/04.e.twentyQuestions/ResponseBuilder.cs:143]: The previous version was using a `private static Random RANDOM = new Random()` instance. I don't know what the threading model is of these teams bots, but if there's any chance that multiple threads could concurrently end up calling this GetRandomResponse method, that would not be safe: like most .NET types, Random is not safe to be used by multiple threads concurrently and risks being corrupted with such use. Instead, the change switches to using the instance returned from `Random.Shared`, which _is_ thread-safe, so this is just switching from using a non-thread-safe instance to a thread-safe instance.  This line is saying "give me a random value in the range [0, response.Length), and return the value from responses at that index".

**Review by singhk97 — APPROVED** [review]

**Comment by singhk97** [comment]
Thanks @stephentoub

**Pull request merged**
